scipy/stats/_pagel.py

```diff
@@ -0,0 +1,466 @@
+
+    >>> from scipy.stats import pagel
+    >>> arranged_ranks = ranks[:, ::-1]
+    >>> res = pagel(arranged_ranks, method='asymptotic')
+    >>> res
+        method: 'asymptotic'
+        pvalue: 0.0012693433690751756
+     statistic: 133.5
+
+    The value of the :math:`L` statistic, 133.5, is as expected:
+
+    >>> import numpy as np
+    >>> m, n = arranged_ranks.shape
+    >>> predicted_ranks = np.arange(n, 0, -1)
+    >>> L = (predicted_ranks * np.sum(arranged_ranks, axis=0)).sum()
+    >>> res.statistic == L
+    True
+
+    The *p*-value is the survival function of the normal distribution evaluated
+    at the standardized test statistic:
+
+    >>> from scipy.stats import norm
+    >>> E0 = (m*n*(n+1)**2)/4
+    >>> V0 = (m*n**2*(n+1)*(n**2-1))/144
+    >>> Lambda = (L-E0)/np.sqrt(V0)
+    >>> p = norm.sf(Lambda)
+    >>> res.pvalue == p
+    True
+
+    This value indicates that there is a 0.1269% chance that
+    the :math:`L` statistic would reach such an extreme value under the null
+    hypothesis. Because 0.1269% is less than 1%, we have evidence to reject
+    the null hypothesis in favor of our alternative at a 99% confidence level.
+
+    Note that the we can also pass in the data as originally tabulated if we
+    provide ``ranked`` and ``predicted_ranks`` arguments. We pass
+    ``ranked=False`` because the originally tabulated data is raw, not ranked,
+    and we pass ``predicted_ranks=[1, 2, 3]``, the ranks we predict for
+    each column of the original table.
+
+    >>> res = pagel(table,                      # data as originally tabulated
+    ...             ranked=False,               # original table is not ranked
+    ...             predicted_ranks=[1, 2, 3],  # we predict increasing rank
+    ...             method="asymptotic"
+    ...             )
+    >>> res
+        method: 'asymptotic'
+        pvalue: 0.0012693433690751756
+     statistic: 133.5
+
+    As presented in [3]_, the *p*-value was calculated based on the asymptotic
+    distribution of the :math:`L` statistic. However, the asymptotic
+    distribution is not very accurate, nor conservative, for :math:`m \leq 12`
+    and :math:`n \leq 8`. Rather than passing in ``method="asymptotic"``,
+    we can allow ``pagel`` to select the appropriate method.
+
+    >>> res = pagel(table, ranked=False, predicted_ranks=[1, 2, 3])
+    >>> res
+        method: 'exact'
+        pvalue: 0.0018191161948127822
+     statistic: 133.5
+
+    Note that ``pagel`` chose to use ``method='exact'`` based on the dimensions
+    of the table and the recommendations in Page's original paper [1]_.
+    This exact *p*-value is greater than that given by the
+    asymptotic approach, but we can still reject the null hypothesis in favor
+    of our alternative at the 99% confidence level.
+    """
+
+    # Possible values of the method parameter and the corresponding function
+    # used to evaluate the p value
+    method = method.lower()
```

Comment:
Now it's documented behavior.

Reply:
But why make it case insensitive?  It is an extra line of a code, and an extra couple words in the docstring, that no one will need.  Let's just drop it.

Reply:
Because I've appreciated this sort of convenience in the past, and it is _only_ one line of code. But again, there are bigger fish to fry, so it's out.